modules/auxiliary/gather/thinmanager_traversal_download.rb

```diff
@@ -0,0 +1,147 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Auxiliary::Report
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'ThinManager Path Traversal (CVE-2023-27856) Arbitrary File Download',
+        'Description' => %q{
+          This module exploits a path traversal vulnerability (CVE-2023-27856) in ThinManager <= v13.0.1 to retrieve arbitrary files from the system.
+
+          The affected service listens by default on TCP port 2031 and runs in the context of NT AUTHORITY\SYSTEM.
+        },
+        'Author' => [
+          'Michael Heinzl', # MSF Module
+          'Tenable' # Discovery and PoC
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2023-27856'],
+          ['URL', 'https://www.tenable.com/security/research/tra-2023-13'],
+          ['URL', 'https://rockwellautomation.custhelp.com/app/answers/answer_view/a_id/1138640']
+        ],
+        'DisclosureDate' => '2023-04-05',
+        'DefaultOptions' => {
+          'RPORT' => 2031,
+          'SSL' => 'False'
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('FILE', [false, 'The file to read from the target system.', '/Windows/win.ini']),
+        OptInt.new('DEPTH', [ true, 'The traversal depth. The FILE path will be prepended with ../ * DEPTH', 7 ])
+      ]
+    )
+  end
+
+  def check
+    begin
+      connect
+    rescue Rex::ConnectionTimeout
+      print_error("Connection to #{datastore['RHOSTS']}:#{datastore['RPORT']} failed.")
+      return CheckCode::Unreachable
```

Comment:
```suggestion
      return CheckCode::Unknown
```
This is why the module is failing msftidy:
```--- Checking new and changed module syntax with tools/dev/msftidy.rb ---
modules/auxiliary/gather/thinmanager_traversal_download.rb - [ERROR] Unrecognized checkcode: Unreachable
```

Seems like `Unknown` is the most accurate of the check codes for this situation: https://docs.metasploit.com/docs/development/developing-modules/guides/how-to-write-a-check-method.html#check-codes

Reply:
Thank you for that, msftidy (and rubocop) didn't show me any issues:
```
$ ../../../tools/dev/msftidy.rb thinmanager_traversal_download.rb 
thinmanager_traversal_download.rb - [*] Rubocop not required for older modules skipping. If making a large update - run rubocop rubocop -a thinmanager_traversal_download.rb and verify all issues are resolved
```
I've probably messed up part of my dev environment some time ago. I'll have to set up a new clean environment..

An updated version with a fixed `CheckCode` will be pushed in a moment.